service/middleware.go

```diff
@@ -119,8 +119,21 @@ func (uis *UIServer) GetSettings() evergreen.Settings {
 // requireUser takes a request handler and returns a wrapped version which verifies that requests
 // request are authenticated before proceeding. For a request which is not authenticated, it will
 // execute the onFail handler. If onFail is nil, a simple "unauthorized" error will be sent.
-func requireUser(onSuccess, onFail http.HandlerFunc) http.HandlerFunc {
+// skipWithToggle indicates that we will skip the authentication if PartialRouteAuthDisabled flag
+// is set. This is a temporary parameter that is in place while we determine if adding auth to all
+// routes will affect users.
+func requireUser(skipWithToggle bool, onSuccess, onFail http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if skipWithToggle {
+			flags, err := evergreen.GetServiceFlags()
+			if err != nil {
+				gimlet.WriteResponse(w, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "retrieving admin settings")))
+			}
+			if flags.PartialRouteAuthDisabled {
+				onFail(w, r)
```

Comment:
Do we need to do the `if onFail != nil` check like below?

Reply:
I'm also not certain why this isn't just returning onSuccess if flags.PartialRouteAuthDisabled is false

Reply:
Yes, I had it as `onSuccess` this morning but somehow convinced myself it should be the other way around 🤦.

Changed it back to 
`if !flags.PartialRouteAuthDisabled {
				onSuccess(w, r)
				return
			}`